internal/services/keyvault/key_vault_certificate_resource.go

```diff
@@ -596,25 +628,38 @@ func resourceKeyVaultCertificateUpdate(d *schema.ResourceData, meta interface{})
 	return resourceKeyVaultCertificateRead(d, meta)
 }
 
-func keyVaultCertificateCreationRefreshFunc(ctx context.Context, client *keyvault.BaseClient, keyVaultBaseUrl string, name string) pluginsdk.StateRefreshFunc {
+func keyVaultCertificateCreationRefreshFunc(ctx context.Context, client *keyvault.BaseClient, keyVaultBaseUrl string, name string, requestID string) pluginsdk.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		res, err := client.GetCertificate(ctx, keyVaultBaseUrl, name, "")
+		operation, err := client.GetCertificateOperation(ctx, keyVaultBaseUrl, name)
 		if err != nil {
-			return nil, "", fmt.Errorf("issuing read request in keyVaultCertificateCreationRefreshFunc for Certificate %q in Vault %q: %s", name, keyVaultBaseUrl, err)
+			return nil, "", fmt.Errorf("failed to read CertificateOperation in keyVaultCertificateCreationRefreshFunc for Certificate %q in Vault %q: %s", name, keyVaultBaseUrl, err)
+		}
+		if operation.RequestID == nil {
+			return nil, "", fmt.Errorf("missing request_id in certificate operation")
+		}
+
+		// If another process starts a new certificate creation right after our
+		// operation was completed we might not observe the completed state. Also as
+		// soon someone started a new create operation there is no way to determine
+		// what happened to the previous operation. Because of that we return an error
+		// if there is a new operation (not the same request ID) running.
+		if *operation.RequestID != requestID {
```

Comment:
@mbfrahry to be honest I can't remember the exact details. i admit this is a hack but as far as I remember it is not possible to get the outcome of the one specific certificate generation. you can only get the operation on that certificate in general.

Reply:
With that said, can we just remove this then? I think we want the most recent certificate operation rather than a specific one

Reply:
@mbfrahry As far as I understand the KeyVault API you can't wait for a specific operation. The [GetCertificateOperation](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/keyvault/v7.0/keyvault#BaseClient.GetCertificateOperation) only takes the certificate name. So I assume you only get the latest running operation.
If  someone starts a new creation of a certificate (new operation) we then would observe the outcome of this action instead of our create operation and this other action could use different parameters (e.g. subjectAlternativeNames etc).
I guess it's quite unlikely that this happens, because if terraform is managing the certificate propably no one lese should run operations on it or at least not at the same time.

Reply:
I see what you're saying but this will cause issues with Terraform because Terraform is getting the latest version of the certificate during the Read function which should be whatever the latest operation is waiting on and not a specific operation (even though as you say, it will most likely be this specific operation).

We should remove this chunk of code and just wait for the latest operation to finish so Terraform has the latest certificate's information as that is what is being set in state

Reply:
See https://github.com/hashicorp/terraform-provider-azurerm/pull/20627#issuecomment-1669338148